[raysonchia] iP

raysonchia opened the pull request:
## Podz

> Bot good.

### Introduction
***
This chatbot will free your mind.
- text-based
- easy
- ~~SLOW~~ FAST to use

All you need:

1. download [here](https://github.com/raysonchia/ip/releases)
2. add your tasks like a *boss*
3. :tada:

And it's **FREEEE**.

Here's the `main` method:
```java
public static void main(String[] args) {
    Podz podz = new Podz();
    podz.run();
}
```
### Usage
***
You may add:
- [x] todos
- [x] deadlines
- [x] events

Adding tasks:
```
todo read book
deadline return book /by 2024-01-05 1800
event project meeting /from 2024-02-01 1200 /to 2024-02-01 1600
```
Output:
```
...
...
____________________________________________________________
Got it. I've added this task:
[E][ ] project meeting (from: February 01 2024 12:00 to: February 01 2024 16:00)
Now you have 3 tasks in the list.
____________________________________________________________
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

raysonchia committed:
Add new bot name, greeting and exit message

raysonchia committed:
Add echo feature

Chatbot will echo input given by user, and exit when user types "bye".

raysonchia committed:
Add ability to store user input to list

User's input is stored and can be displayed using the "list" command.

raysonchia committed:
Add Task class

raysonchia committed:
Add ability to mark/unmark tasks

User may mark/unmark tasks in list using commands "mark" and "unmark".

raysonchia committed:
Add Deadline class

raysonchia committed:
Add feature to specify tasks with deadline

Tasks with deadlines (e.g. dates) can be added to list.

raysonchia committed:
Add Todo class

raysonchia committed:
Add feature to add Todo tasks to list

Todos are tasks without any date/time attached.

raysonchia committed:
Add Event class

raysonchia committed:
Add spacing

raysonchia committed:
Add event tasks

Event tasks with a specific starting and ending time can be added to list.

raysonchia committed:
Add automated text UI testing

raysonchia committed:
Add error handling

Program will deal with invalid inputs entered by the user.

raysonchia committed:
Add delete feature

User may delete tasks from lists.

raysonchia committed:
Update automated testing

Add previously untested features (e.g. delete) and error handling.

raysonchia committed:
Update commands from strings to enums

raysonchia committed:
Add saving and loading task from hard disk

raysonchia committed:
Merge branch 'branch-Level-7'

raysonchia committed:
Add recognition of dates and times

When creating deadline and event tasks, user can specify date and time
in the format yyyy-mm-dd HHmm.
For example: deadline return book /by 2024-01-05 1800
The time parameter is optional.

raysonchia committed:
Merge branch 'branch-Level-8'

raysonchia committed:
Refactor saving/loading operations to Storage class

raysonchia committed:
Refactor task list operations

raysonchia committed:
Refactor into Parser and Ui class

raysonchia committed:
Merge branch 'branch-A-MoreOOP'

raysonchia committed:
Organize files into packages

raysonchia committed:
Merge branch 'branch-A-Packages'

raysonchia committed:
Merge remote-tracking branch 'origin/add-gradle-support'

raysonchia committed:
Configure gradle build file

teojunda commented on src/main/java/podz/storage/Storage.java:
```diff
@@ -0,0 +1,90 @@
+        Storage.FILE_PATH = filePath;
+    }
+
+    public static void updateSaved(ArrayList<Task> AL) {
```
Perhaps it would be clearer to rename AL to tasks?

teojunda commented on src/main/java/podz/task/Event.java:
```diff
@@ -0,0 +1,50 @@
+        this.from = from;
+        this.to = to;
+        
+        String fromDT[] = from.split(" ");
+        String toDT[] = to.split(" ");
```
Perhaps the array specifiers should be attached to the type instead of the variable.

teojunda commented on src/main/java/podz/task/Deadline.java:
```diff
@@ -0,0 +1,42 @@
+        super(description);
+        this.by = by;
+
+        String dateAndTime[] = by.split(" ");
```
Perhaps the array specifiers should be attached to the type instead of the variable.

raysonchia replied:
Thanks for your review! I appreciate the feedback and have addressed the suggested changes.

teojunda reviewed (commented):
Great job 👍 
I really like how readable your code was. However, I feel that there are a few variables that can be renamed to improve the code clarity.

raysonchia committed:
Remove unnecessary try-catch

Let parser catch the exception.

raysonchia committed:
Add JUnit tests

Made some tests for the DeleteCommand and MarkCommand execute() method.

raysonchia committed:
Merge branch 'branch-A-JUnit'

raysonchia committed:
Edit build name

raysonchia committed:
Merge branch 'branch-A-Jar'

raysonchia committed:
Add JavaDoc comments

raysonchia committed:
Fix some Java coding standard violations

raysonchia committed:
Add find command to filter tasks by keyword

raysonchia committed:
Merge branch 'branch-A-JavaDoc'

raysonchia committed:
Merge branch 'branch-A-CodingStandard'

raysonchia committed:
Merge branch 'branch-Level-9'

raysonchia committed:
Integrate the checkstyle plugin

raysonchia committed:
Fix key checkstyle violations

raysonchia committed:
Merge branch 'branch-A-CheckStyle'

raysonchia committed:
Remove text based formatting related code

raysonchia committed:
Add GUI

raysonchia committed:
Merge branch 'branch-Level-10'

YuZizhen commented on src/main/java/podz/parser/Parser.java:
```diff
@@ -5,6 +5,7 @@
 import podz.commands.DeadlineCommand;
 import podz.commands.DeleteCommand;
 import podz.commands.EventCommand;
+import podz.commands.FindCommand;
```
Nice to put the find command as a seperate class. This makes the code more readable!

YuZizhen commented on src/main/java/podz/parser/Parser.java:
```diff
@@ -196,4 +200,17 @@ private Command prepareDelete(String[] inputs) {
+        } catch (PodzException e) {
+            return new IncorrectCommand(e);
+        }
+    }
```
Good to put the helper function as a private function!

YuZizhen commented on src/main/java/podz/task/TaskList.java:
```diff
@@ -86,6 +86,25 @@ public String getListCounter() {
+
+        return toString(filteredTasks);
+    }
+
```
Great! It is clearer to have this isRelavantTask!

YuZizhen reviewed (commented):
Nice implementation that is clear and easy to understand

raysonchia committed:
Remove unused code

raysonchia committed:
Add assertions

There are no assertions in the program.

Absence of assertions makes it more difficult to detect mistake in code.

Add assertions in various points of the code to verify our assumptions.

Using assertions help to address the application's correctness.

raysonchia committed:
Merge pull request #1 from raysonchia/branch-A-Assertions

Add assertions

raysonchia committed:
Improve code quality

There are several areas where code does not follow best practices.

They can be improved to enhance code readability.

Refactor of several files. Adhere to practices such as SLAP, avoiding
deep nesting, complicated expressions and magic numbers.

raysonchia committed:
Merge pull request #2 from raysonchia/branch-A-CodeQuality

Improve code quality

raysonchia committed:
Rearrange order of method

raysonchia committed:
Refactor findTasks() to use streams

raysonchia committed:
Merge pull request #3 from raysonchia/branch-A-Streams

Refactor findTasks() to use streams

raysonchia committed:
Add empty message guard clause

raysonchia committed:
Add command flexibility with shorter aliases

raysonchia committed:
Merge branch 'branch-BCD-Extension'

raysonchia committed:
Add JavaDoc and perform code cleaning

raysonchia committed:
Allow resizing of window and its contents

raysonchia committed:
Update .gitignore

raysonchia committed:
Adjust top bar position and padding

raysonchia committed:
Update font style in GUI

raysonchia committed:
Merge branch 'branch-A-BetterGui'

raysonchia committed:
Add code credit

raysonchia committed:
Add missing JavaDoc punctuation

raysonchia committed:
Update profile pictures

raysonchia committed:
Adjust padding and image dimensions

raysonchia committed:
Update command responses with personality

raysonchia committed:
Update variable and function names

raysonchia committed:
Adjust padding

raysonchia committed:
Merge branch 'branch-A-Personality'

raysonchia committed:
Remove unused resources

raysonchia committed:
Add error checking for date format

raysonchia committed:
Add name to application window bar

raysonchia committed:
Add space

raysonchia committed:
Add GUI window screenshot

raysonchia committed:
Update user guide

raysonchia committed:
Remove extra words

raysonchia committed:
Update main class

raysonchia committed:
Update repo's README

raysonchia committed:
Update credits link

raysonchia committed:
Update default window dimension
